tools/convertUsageStatsLogFile.php

```diff
@@ -0,0 +1,570 @@
+            $newLogEntry = json_encode($newEntry) . PHP_EOL;
+            file_put_contents($newFilePath, $newLogEntry, FILE_APPEND);
+            $isSuccessful = true;
+        }
+        fclose($fhandle);
+
+        if ($isSuccessful) {
+            $renameFilePath = $archivePath . '/' . $path_parts['filename'] . '_old.log';
+            if (!rename($filePath, $renameFilePath)) {
+                echo "Cound not rename the file {$filePath} to {$renameFilePath}.\n";
+                exit(1);
+            } else {
+                echo "Old file was renamed to {$renameFilePath}.\n";
+            }
+            if (!rename($newFilePath, $filePath)) {
+                echo "Cound not rename the new file {$newFilePath} to {$filePath}.\n";
+                exit(1);
+            } else {
+                echo "File {$filePath} is converted.\n";
+            }
+            if ($extension == 'gz') {
+                try {
+                    $renameFilePath = $fileMgr->gzCompressFile($renameFilePath);
+                    $filePath = $fileMgr->gzCompressFile($filePath);
+                } catch (Exception $e) {
+                    printf($e->getMessage() . "\n");
+                    exit(1);
+                }
+            }
+        } else {
+            echo "File {$filePath} could not be successfully converted.\n";
+            exit(1);
+        }
+    }
+
+    /**
+     * Get data from the passed log entry.
+     */
+    private function getDataFromLogEntry(string $entry): array
+    {
+        $entryData = [];
+        if (preg_match(self::PARSEREGEX, $entry, $m)) {
+            $associative = count(array_filter(array_keys($m), 'is_string')) > 0;
+            $entryData['ip'] = $associative ? $m['ip'] : $m[1];
+            $entryData['date'] = strtotime($associative ? $m['date'] : $m[2]);
+            $entryData['url'] = urldecode($associative ? $m['url'] : $m[3]);
+            $entryData['returnCode'] = $associative ? $m['returnCode'] : $m[4];
+            $entryData['userAgent'] = $associative ? $m['userAgent'] : $m[5];
+        }
+        return $entryData;
+    }
+
+    /**
+     * Validate a access log entry.
+     * This maybe does not have much sense, but because it was used till now, we will leave it.
+     */
+    private function isLogEntryValid(array $entry): bool
+    {
+        if (empty($entry)) {
+            return false;
+        }
+        $date = $entry['date'];
+        if (!is_numeric($date) && $date <= 0) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Get assoc type, page, operation and args from the passed url,
+     * if it matches any that's defined in getExpectedPageAndOp().
+     */
+    private function getUrlMatches(string $url, int $lineNumber): array
+    {
+        $noMatchesReturner = [null, null, null, null, null];
+
+        $expectedPageAndOp = $this->getExpectedPageAndOp();
+
+        $pathInfoDisabled = Config::getVar('general', 'disable_path_info');
```

Comment:
FYI, the `disable_path_info` option has been marked deprecated since 3.2.0 and I am ready to remove it. It's likely not working well (e.g. with API requests) anyway. If you find yourself doing a lot of coding to accommodate the feature, let me know and I'll prioritize its removal instead!

Reply:
Are the functions Core::getContextPaths, getPage, getOp and getArtgs going to change? Will they still have/consider `$isPathInfo` as parameter? If so I do not have to change much -- I would add a variable at the beginning of the script so that the user can adapt it if the log file contains such URLs. If not I would need to 'copy' those function here in the script...

Reply:
Since you're not using the 3rd parameter to `Core::_getUserVar`, I'd recommend just using `parse_url` directly, e.g. for the two use cases of `Core::_getUserVar` in this PR:
```php
parse_str((string) parse_url($urlInfo, PHP_URL_QUERY), $userVars);
$contextPaths[$key] = $userVars[$contextName] ?? null;
```
...and...
```php
parse_str((string) parse_url($urlInfo, PHP_URL_QUERY), $userVars);
$component = $userVars[$varName] ?? null;
```

Reply:
Oh yes, that makes sense! :+1: